src/backend/core/recording/worker/factories.py

```diff
@@ -0,0 +1,83 @@
+"""Factory, configurations and Protocol to create worker services"""
+
+import logging
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import Any, ClassVar, Dict, Optional, Protocol, Type
+
+from django.conf import settings
+from django.utils.module_loading import import_string
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class WorkerServiceConfig:
+    """Declare Worker Service common configurations"""
+
+    output_folder: str
+    server_configurations: Dict[str, Any]
+    verify_ssl: Optional[bool]
+    bucket_args: Optional[dict]
+
+    @classmethod
+    @lru_cache
+    def from_settings(cls) -> "WorkerServiceConfig":
+        """Load configuration from Django settings with caching for efficiency."""
+
+        logger.debug("Loading WorkerServiceConfig from settings.")
+        return cls(
+            output_folder=settings.RECORDING_OUTPUT_FOLDER,
+            server_configurations=settings.LIVEKIT_CONFIGURATION,
+            verify_ssl=settings.RECORDING_VERIFY_SSL,
+            bucket_args={
+                "endpoint": settings.AWS_S3_ENDPOINT_URL,
+                "access_key": settings.AWS_S3_ACCESS_KEY_ID,
+                "secret": settings.AWS_S3_SECRET_ACCESS_KEY,
+                "region": settings.AWS_S3_REGION_NAME,
+                "bucket": settings.AWS_STORAGE_BUCKET_NAME,
+                "force_path_style": True,
+            },
+        )
+
+
+class WorkerService(Protocol):
+    """Define the interface for interacting with a worker service."""
+
+    hrid: ClassVar[str]
+
+    def __init__(self, config: WorkerServiceConfig):
+        """Initialize the service with the given configuration."""
+
+    def start(self, room_id: str, recording_id: str) -> str:
+        """Start a recording for a specified room."""
+
+    def stop(self, worker_id: str) -> str:
+        """Stop recording for a specified worker."""
+
+
+@lru_cache(maxsize=1)
+def get_worker_registry() -> Dict[str, Type[WorkerService]]:
+    """Return a cached mapping of human-readable IDs to worker service classes."""
+
+    registry: Dict[str, Type[WorkerService]] = {}
+
+    for class_path in settings.RECORDING_WORKER_CLASSES:
+        worker_class = import_string(class_path)
+        registry[worker_class.hrid] = worker_class
+
+    return registry
+
+
+def get_worker_service(hrid: str) -> WorkerService:
+    """Instantiate a worker service by its human-readable identifier."""
+
+    worker_service_registry = get_worker_registry()
+    default_config = WorkerServiceConfig.from_settings()
+
+    worker_service_cls = worker_service_registry.get(hrid)
+
+    if not worker_service_cls:
+        raise ValueError(f"Unknown worker service for hrid: {hrid}.")
+
+    return worker_service_registry[hrid](config=default_config)
```

Comment:
initially I created a proper factory class, discussed it IRL with @sampaccoud which convinced me to follow this approach. Let's discuss it IRL.

Reply:
I have refactored the factory function to something more flexible. Closing this discussion